### DISCUSSION TITLE
Zoom Monitor Fail

### DISCUSSION THREAD

**Discussion by FlorianGrabowski** [tryptech/obs-zoom-and-follow#14] [General]
no matter how I configure it the script always takes the main display as a source for the zoom coordinates not the one from "Select monitor to use base for zooming" nor the one from the selected display source. Am I doing something wrong ?

**Comment by FlorianGrabowski** [comment]
can somone tell me if there is a way to play with y offset or the monitor index to get the mousetracking on my secondarry screen?

**Comment by tryptech** [comment]
Could you tell me a little more the system you're running it on? 
How many monitors and their resolutions, what operating system, etc.

**Comment by FlorianGrabowski** [comment]
never mind you can close this Disscusson. I got a workaround by changing some lines in the script.

**Reply by ngaerlan** [reply]
how did you fix it, what did you change in the script? pls, thx

**Reply by FlorianGrabowski** [reply]
I changed the following Lines in the script. (im not a python guy so forgive me my crapy workaround and bad (no) exeption handling) 

I´m on Windows and wanted to use my 2. screen for everything it is configured to be on the right of the main screen thats important because of the mouse position calculation in line 96. If you want to use te script for your 1. and 2. screen you have to configure both versions of the script the original and my workaround on the apropriate sources and shortcuts.

![000007](https://user-images.githubusercontent.com/47364896/118765445-62064b80-b87b-11eb-9e19-7e8a66b69f08.jpg)

![000008](https://user-images.githubusercontent.com/47364896/118765447-629ee200-b87b-11eb-80fb-086527a4504d.jpg)

![000009](https://user-images.githubusercontent.com/47364896/118765436-5dda2e00-b87b-11eb-8ae4-b7fc851b9f63.jpg)

all lines of code canged to copy:
```python
monitor_idx = 1
```
```python
x_o = (mousePos[0] - get_monitors()[0].width) 
```
```python
obs.obs_data_set_default_string(settings, "Monitor", get_monitors()[1].name)
obs.obs_data_set_default_int(settings, "Manual Monitor Width", get_monitors()[1].width)
obs.obs_data_set_default_int(settings, "Manual Monitor Height", get_monitors()[1].height)
```

Surely there is a posibillity to integrate these changes into the script properly with corresponding UI but I dont know how to do this. Maby @tryptech integrates it in future paches. Btw. great script thx verry helpfull!

**Reply by astronaut954** [reply]
Hi, @FlorianGrabowski, I need something similar to what you did, I need to set 2 scripts, one for my main monitor (Monitor 0) and one for my second monitor (Monitor 1). I've edited the script just like you did, but when the mouse will follow the Y position of the first monitor and not the second one. The Main Monitor uses 1920 x 1080 and the Second Monitor uses 1280 x 800. Can you give me some help?

**Reply by FlorianGrabowski** [reply]
You should have a look at line 97 where y_o gets defined. There you can test something like apending ```-(monitorHeightDiff)```.
